### PR TITLE
Find identical levels for variables to fix GFS-FV3 update

### LINKED ISSUE #422: Fix builds
We're currently broken in two of the notebooks, I sense due to the GFS FV3 update, but I'm not positive:

notebooks/MetPy_Advanced/Isentropic Analysis.ipynb:
```pytb
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-11-47590c238f43> in <module>
      1 isen_level = np.array([320]) * units.kelvin
      2 isen_press, isen_u, isen_v = mpcalc.isentropic_interpolation(isen_level, press,
----> 3                                                              temperature, u, v)
~/miniconda/envs/unidata/lib/python3.7/site-packages/metpy/xarray.py in wrapper(*args, **kwargs)
    434         kwargs = {name: (v.metpy.unit_array if isinstance(v, xr.DataArray) else v)
    435                   for name, v in kwargs.items()}
--> 436         return func(*args, **kwargs)
    437     return wrapper
    438 
~/miniconda/envs/unidata/lib/python3.7/site-packages/metpy/units.py in wrapper(*args, **kwargs)
    303                                 '    x = x * units.meter / units.second')
    304                     raise ValueError(msg)
--> 305                 return func(*args, **kwargs)
    306 
    307             return wrapper
~/miniconda/envs/unidata/lib/python3.7/site-packages/metpy/calc/thermo.py in isentropic_interpolation(theta_levels, pressure, temperature, *args, **kwargs)
   1581     slices[axis] = slice(None)
   1582     slices = tuple(slices)
-> 1583     pres = np.broadcast_to(pres[slices], temperature.shape) * pres.units
   1584 
   1585     # Sort input data
~/miniconda/envs/unidata/lib/python3.7/site-packages/numpy/lib/stride_tricks.py in broadcast_to(array, shape, subok)
    180            [1, 2, 3]])
    181     """
--> 182     return _broadcast_to(array, shape, subok=subok, readonly=True)
    183 
    184 
~/miniconda/envs/unidata/lib/python3.7/site-packages/numpy/lib/stride_tricks.py in _broadcast_to(array, shape, subok, readonly)
    127     it = np.nditer(
    128         (array,), flags=['multi_index', 'refs_ok', 'zerosize_ok'] + extras,
--> 129         op_flags=[op_flag], itershape=shape, order='C')
    130     with it:
    131         # never really has writebackifcopy semantics
ValueError: operands could not be broadcast together with remapped shapes [original->remapped]: (31,1,1) and requested shape (34,101,161)
```

notebooks/Model_Output/Downloading model fields with NCSS.ipynb:
```pytb
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-13-bc5c5856934c> in <module>
      1 import metpy.calc as mpcalc
      2 
----> 3 Td = mpcalc.dewpoint_rh(T, relh)
~/miniconda/envs/unidata/lib/python3.7/site-packages/metpy/xarray.py in wrapper(*args, **kwargs)
    434         kwargs = {name: (v.metpy.unit_array if isinstance(v, xr.DataArray) else v)
    435                   for name, v in kwargs.items()}
--> 436         return func(*args, **kwargs)
    437     return wrapper
    438 
~/miniconda/envs/unidata/lib/python3.7/site-packages/metpy/units.py in wrapper(*args, **kwargs)
    303                                 '    x = x * units.meter / units.second')
    304                     raise ValueError(msg)
--> 305                 return func(*args, **kwargs)
    306 
    307             return wrapper
~/miniconda/envs/unidata/lib/python3.7/site-packages/metpy/calc/thermo.py in dewpoint_rh(temperature, rh)
    698     if np.any(rh > 1.2):
    699         warnings.warn('Relative humidity >120%, ensure proper units.')
--> 700     return dewpoint(rh * saturation_vapor_pressure(temperature))
    701 
    702 
~/miniconda/envs/unidata/lib/python3.7/site-packages/pint/quantity.py in __mul__(self, other)
    812 
    813     def __mul__(self, other):
--> 814         return self._mul_div(other, operator.mul)
    815 
    816     __rmul__ = __mul__
~/miniconda/envs/unidata/lib/python3.7/site-packages/pint/quantity.py in wrapped(self, *args, **kwargs)
     48 def reduce_dimensions(f):
     49     def wrapped(self, *args, **kwargs):
---> 50         result = f(self, *args, **kwargs)
     51         if result._REGISTRY.auto_reduce_dimensions:
     52             return result.to_root_units()
~/miniconda/envs/unidata/lib/python3.7/site-packages/pint/quantity.py in _mul_div(self, other, magnitude_op, units_op)
    800             other = other.to_root_units()
    801 
--> 802         magnitude = magnitude_op(new_self._magnitude, other._magnitude)
    803         units = units_op(new_self._units, other._units)
    804 
~/miniconda/envs/unidata/lib/python3.7/site-packages/numpy/ma/core.py in __mul__(self, other)
   4096         if self._delegate_binop(other):
   4097             return NotImplemented
-> 4098         return multiply(self, other)
   4099 
   4100     def __rmul__(self, other):
~/miniconda/envs/unidata/lib/python3.7/site-packages/numpy/ma/core.py in __call__(self, a, b, *args, **kwargs)
   1016         with np.errstate():
   1017             np.seterr(divide='ignore', invalid='ignore')
-> 1018             result = self.f(da, db, *args, **kwargs)
   1019         # Get the mask for the result
   1020         (ma, mb) = (getmask(a), getmask(b))
ValueError: operands could not be broadcast together with shapes (31,) (34,) 
```

### PR DESCRIPTION
**Pull request by zbruick**
Update to the Isentropic Analysis and NCSS notebooks to find identical vertical levels between temperature and other variables due to the differing number of vertical levels caused by the FV3 update to the GFS.

Fixes #422.

### CODE REVIEW

**Comment by dopplershift** [comment]
Before saving and committing, make sure you "Clear All Output" under the edit mode. We do that to avoid confusing the learners when we teach.

**Comment by zbruick** [comment]
Makes perfect sense - hopefully got it cleaned up.

File: notebooks/MetPy_Advanced/Isentropic Analysis.ipynb [hunk 1]
```diff
@@ -210,24 +250,37 @@
...
-    "v = ds['v-component_of_wind_isobaric'][0]"
+    "v = ds['v-component_of_wind_isobaric'][0]\n",
+    "\n",
+    "# Due to a different number of vertical levels find where they are common\n",
+    "lev_hght = temperature.coords['isobaric4'] * units.Pa\n",
+    "lev_uwnd = u.coords['isobaric'] * units.Pa\n",
+    "_, _, common_ind = np.intersect1d(lev_uwnd, lev_hght, return_indices=True)\n",
+    "temperature = temperature[common_ind, :, :]\n",
+    "# Reassign units to variables\n",
+    "temperature.attrs['units'] = 'kelvin'\n",
```
Comment (dopplershift) [inline notebooks/MetPy_Advanced/Isentropic Analysis.ipynb]: Is xarray dropping units here or are the original units bad?

**Comment by dopplershift** [comment]
This has added a bunch of pycharm-specific json to the notebook. I'm unsure how to feel about that...

File: notebooks/MetPy_Advanced/Isentropic Analysis.ipynb (hunk 1 above)
Comment (zbruick) [reply notebooks/MetPy_Advanced/Isentropic Analysis.ipynb]: It's dropping units there.

**Comment by zbruick** [comment]
I can create a new branch and edit the notebook just through Jupyter to keep that out if that would be better.

**Comment by dopplershift** [comment]
It looks like this is primarily failing on the build against MetPy master because it was using `lat_lon_grid_spacing` in `notebooks/MetPy_Case_Study/MetPy_Case_Study.ipynb`. Can you go ahead and update to use `lat_lon_grid_deltas`? It should be a straightforward replacement.

File: notebooks/MetPy_Advanced/Isentropic Analysis.ipynb (hunk 1 above)
Comment (dopplershift) [reply notebooks/MetPy_Advanced/Isentropic Analysis.ipynb]: 😞 That one *doesn't* seem sensible.

**Comment by zbruick** [comment]
Updated as requested

**Comment by zbruick** [comment]
Let me know if the errors with Travis are something I need to fix or if those are known issues. They don't look related to the notebooks already updated within this PR.

**Comment by dopplershift** [comment]
Looks like current errors will be fixed by #425.

**Pull request merged**